packages/@rnw-scripts/integrate-rn/src/integrateRN.ts

```diff
@@ -0,0 +1,346 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ */
+
+import * as _ from 'lodash';
+import * as fs from 'fs';
+import * as ora from 'ora';
+import * as path from 'path';
+import * as semver from 'semver';
+import {exec, ExecOptions} from 'child_process';
+import {
+  upgradeOverrides,
+  validateManifest,
+  ValidationError,
+} from 'react-native-platform-override';
+
+import * as yargs from 'yargs';
+
+import {
+  enumerateLocalPackages,
+  findPackage,
+  WritableNpmPackage,
+} from '@rnw-scripts/package-utils';
+
+import findRepoRoot from '@rnw-scripts/find-repo-root';
+
+(async () => {
+  const {argv} = yargs.check(args => {
+    if (args._.length === 1 && semver.valid(args._[0])) {
+      return true;
+    } else {
+      throw new Error('Usage: integrate-rn <version>');
+    }
+  });
+  const version = argv._[0];
+
+  await funcStep(
+    `Updating packages and dependencies to react-native@${version}`,
+    async () => await updateDependencies(version),
+  );
+
+  await funcStep('Upgrading out-of-date overrides', upgradePlatformOverrides);
+  await funcStep(
+    'Performing additional override validation',
+    validatePlatformOverrides,
+  );
+
+  await failableCommandStep('yarn lint:fix');
+  await commandStep('yarn build');
+  await failableCommandStep('yarn lint');
+})();
+
+/**
+ * Enumerate packages for out of-tree platforms
+ */
+async function enumeratePlatformPackages(): Promise<WritableNpmPackage[]> {
+  return await enumerateLocalPackages(async pkg => {
+    try {
+      await fs.promises.access(path.join(pkg.path, 'overrides.json'));
+      return true;
+    } catch {
+      return false;
+    }
+  });
+}
+
+/**
+ * Update all package.json dependencies in the repo to account for a new
+ * version of React Native
+ */
+async function updateDependencies(newVersion: string) {
+  // 1. Cache a copy of the previous RN package.json
+  const origRnPackageJson = (await findPackage('react-native'))!.json;
```

Comment:
This is kind of sketch, since we don't declare an RN dependency and assume something gets hoisted of a single version. Need to update to look from cwd of a platform package, or at least add a comment.